src/Test/Utilities/Portable/CommonTestBase.CompilationVerifier.cs

```diff
@@ -312,10 +297,32 @@ private IModuleSymbol GetModuleSymbolForEmittedImage(ImmutableArray<byte> peImag
 
                 var targetReference = LoadTestEmittedExecutableForSymbolValidation(peImage, _compilation.Options.OutputKind, display: _compilation.AssemblyName);
                 var references = _compilation.References.Concat(new[] { targetReference });
-                var assemblies = _test.ReferencesToModuleSymbols(references, importOptions);
+                var assemblies = GetReferencesToModuleSymbols(references, importOptions);
                 return assemblies.Last();
             }
 
+            private IEnumerable<IModuleSymbol> GetReferencesToModuleSymbols(IEnumerable<MetadataReference> references, MetadataImportOptions importOptions)
+            {
+                var dummy = _compilation
+                    .RemoveAllReferences()
```

Comment:
RemoveAllReferences is just the same as WithReferences with some empty array. What is the benefit of calling WithReferences twice?